llvm/lib/MC/RepoObjectWriter.cpp

```diff
@@ -687,6 +695,10 @@ pstore::index::digest RepoObjectWriter::buildCompilationRecord(
       CompilationHash.update(Name.size());
       CompilationHash.update(stringViewAsRef(Name));
     }
+    // Update the fragment to remember the corrresponding compilation member.
+    if (FragmentPos != Fragments.end())
+      FragmentPos->second.CorrespondingCompilationMember =
+          &CompilationMembers.back();
```

Comment:
Since CompilationMembers is reserved to Tickets.size() and it is always less or equal to Tickets.size(), the emplace_back() will not invalidates pointers (and iterators) into the vector.

Reply:
Ok, that's good: I missed the reserve. I think that we should add an assertion to ensure that this is always true (now and in the future) and to ask as documentation of the precondition.

Reply:
Thanks Paul. An assertion has been added.